### PR TITLE
Refactor of zip plugin

### PR DESCRIPTION
**Pull request by chrisbra**

### CODE REVIEW

File: runtime/autoload/zip.vim [hunk 1]
```diff
@@ -83,29 +83,22 @@ fun! zip#Browse(zipfile)
...
-   if &verbose > 1
-    echoerr "the zip plugin is not available (your vim doesn't support fnameescape())"
-   endif
-   return
-  endif
   if !executable(g:zip_unzipcmd)
    redraw!
-   echohl Error | echo "***error*** (zip#Browse) unzip not available on your system"
-   let &report= repkeep
+   echohl Error | echomsg "***error*** (zip#Browse) unzip not available on your system"
```
Comment (k-takata) [inline runtime/autoload/zip.vim]: How about defining functions to show error or warning messages?

Comment (chrisbra) [reply runtime/autoload/zip.vim]: good idea.

File: runtime/autoload/zip.vim [hunk 2]
```diff
@@ -83,29 +83,22 @@ fun! zip#Browse(zipfile)
...
+   call s:RestoreOpts(dict)
```
Comment (ghost) [inline runtime/autoload/zip.vim]: This could be a good use case of the :defer feature?

Comment (chrisbra) [reply runtime/autoload/zip.vim]: good idea. Have amended by a new commit

**Pull request closed without merging**